component/src/main/java/org/wso2/extension/siddhi/io/jms/source/JMSMessageProcessor.java

```diff
@@ -66,59 +71,139 @@ public boolean receive(CarbonMessage carbonMessage, CarbonCallback carbonCallbac
-                String event = ((TextCarbonMessage) carbonMessage).getText();
+            if (message instanceof TextMessage) {
+                String[] transportProperties = populateTransportHeaders(message);
+                String event = ((TextMessage) message).getText();
                 sourceEventListener.onEvent(event, transportProperties);
-            } else if (carbonMessage.getClass() == MapCarbonMessage.class) {
-                String[] transportProperties = populateTransportHeaders(carbonMessage);
+            } else if (message instanceof MapMessage) {
+                String[] transportProperties = populateTransportHeaders(message);
                 Map<String, String> event = new HashMap<>();
-                MapCarbonMessage mapCarbonMessage = (MapCarbonMessage) carbonMessage;
-                Enumeration<String> mapNames = mapCarbonMessage.getMapNames();
+                MapMessage mapEvent = (MapMessage) message;
+                Enumeration<String> mapNames = mapEvent.getMapNames();
                 while (mapNames.hasMoreElements()) {
                     String key = mapNames.nextElement();
-                    event.put(key, mapCarbonMessage.getValue(key));
+                    event.put(key, mapEvent.getString(key));
                 }
                 sourceEventListener.onEvent(event, transportProperties);
+            } else if (message instanceof ByteBuffer) {
+                String[] transportProperties = populateTransportHeaders(message);
+                sourceEventListener.onEvent(message, transportProperties);
             } else {
-                throw new JMSInputAdaptorRuntimeException("The message type of the JMS message" +
-                                                                  carbonMessage.getClass() + " is not supported!");
+                throw new JMSInputAdaptorRuntimeException("The message type of the JMS message"
+                        + message.getClass() + " is not supported!");
             }
             // ACK only if the event is processed i.e: no exceptions thrown from the onEvent method.
-            if (carbonCallback != null) {
-                carbonCallback.done(carbonMessage);
+            if (jmsCallback != null) {
+                jmsCallback.done(true);
             }
-        } catch (RuntimeException e) {
+        } catch (JMSConnectorException | JMSException e) {
             throw new JMSInputAdaptorRuntimeException("Failed to process JMS message.", e);
         }
-        return true;
     }
 
-    private String[] populateTransportHeaders(CarbonMessage carbonMessage) {
-        if (requestedTransportPropertyNames.length > 0) {      //cannot be null according to siddhi impl
+    private String[] populateTransportHeaders(Message message) throws JMSException, JMSConnectorException {
+        if (requestedTransportPropertyNames.length > 0) {
+            //cannot be null according to siddhi impl
+            message.getPropertyNames();
```

Comment:
What is the purpose of this line? We are not assigning the return value to anything here...

Reply:
Fixed with commit [#3fb99a6822b8f5bb32d97ebdc4b7495aa01a0dfc](https://github.com/wso2-extensions/siddhi-io-jms/pull/25/commits/3fb99a6822b8f5bb32d97ebdc4b7495aa01a0dfc)